tests/test_text_models.py

```diff
@@ -0,0 +1,409 @@
+import json
+import unittest
+from pathlib import Path
+
+from mlx_engine.logging import log_info
+from .utils import model_getter, model_load_and_tokenize_prompt, read_text_file
+from mlx_engine.generate import (
+    load_model,
+    load_draft_model,
+    is_draft_model_compatible,
+    unload_draft_model,
+    tokenize,
+    create_generator,
+)
+
+
+class TestTextModels(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        """Set up test resources that will be shared across all test methods"""
+        cls.model_path_prefix = Path("~/.cache/lm-studio/models").expanduser().resolve()
+        cls.test_data_dir = Path(__file__).parent / "data"
+
+    def test_repetition_penalty_applies(self):
+        model_path = model_getter("lmstudio-community/Qwen2.5-0.5B-Instruct-MLX-8bit")
+        model_kit = load_model(model_path=model_path, max_kv_size=4096)
+        prompt = """<|im_start|>user
+The quick brown fox jumped over the lazy dog. The quick brown fox jumped over the lazy dog. The quick brown fox jumped over the lazy dog. Repeat what I said.
+<|im_end|>\n<|im_start|>assistant\n"""
+        prompt_tokens = tokenize(model_kit, prompt)
+        generated_text = ""
+
+        def generate() -> None:
+            nonlocal generated_text
+            for result in create_generator(
+                model_kit=model_kit,
+                prompt_tokens=prompt_tokens,
+                repetition_penalty=3,  # this will make it so that model shouldn't repeat. If set to 0, it will
+                repetition_context_size=64,
+                seed=0,
+                max_tokens=20,
+                temp=0.0,
+            ):
+                print(result.text, end="", flush=True)
+                generated_text += result.text
+                if result.stop_condition:
+                    break
+            print("\n", flush=True)
+
+        generate()
+        self.assertGreater(len(generated_text), 0, "Model failed to generate any text")
+        self.assertNotIn(
+            "The quick brown fox jumped over the lazy dog.", generated_text
+        )
+
+    def test_prompt_caching(self):
+        model_path = model_getter("mlx-community/gemma-3-text-4b-it-4bit")
+        model_kit = load_model(model_path=model_path, max_kv_size=4096)
+        file_content = read_text_file(
+            self.test_data_dir / "ben_franklin_autobiography_start.txt"
+        )
```

Comment:
```suggestion
        file_path = self.test_data_dir / "ben_franklin_autobiography_start.txt"
        file_content = file_path.read_text()
```
and can probably get rid of `read_text_file`

Reply:
Nice, didn't know about this